File: rust/actix/src/modules/account/module.rs
use actix_web::{web, Scope, dev::ServiceFactory};
use actix_web::dev::ServiceRequest;
use actix_web::dev::ServiceResponse;
use actix_web::Error;
use super::controller::AccountController;
use crate::modules::auth::middleware::AuthMiddleware;

pub fn account_module() -> Scope<impl ServiceFactory<ServiceRequest, Config = (), Response = ServiceResponse, Error = Error, InitError = ()>> {
    web::scope("/contas")
        .wrap(AuthMiddleware)
        .service(
            web::resource("")
                .route(web::get().to(AccountController::find_all))
                .route(web::post().to(AccountController::create))
        )
        .service(
            web::resource("/{id}")
                .route(web::get().to(AccountController::find_by_id))
        )
}

File: go/fiber_orm/internal/modules/account/module.go
package account

import (
	auth "tcc/internal/modules/auth/middleware"
	"tcc/internal/modules/branch"
	"tcc/internal/modules/client"
	"tcc/internal/modules/manager"

	"github.com/gofiber/fiber/v2"
)

func AccountModule(router *fiber.App) {
	service := AccountService{
		repository: struct{account AccountRepository; client client.ClientRepository; branch branch.BranchRepository; manager manager.ManagerRepository}{
			account: AccountRepository{},
			client: client.ClientRepository{},
			branch: branch.BranchRepository{},
			manager: manager.ManagerRepository{},
		},
	}
	controller := AccountController{
		service: service,
	}

	group := router.Group("/contas")
	{
		group.Get("", auth.AuthMiddleware(), controller.FindAll)
		group.Post("", auth.AuthMiddleware(), controller.Create)
		group.Get("/:id", auth.AuthMiddleware(), controller.FindByCPF)
	}
}